core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -24,6 +24,20 @@
 #include <cstdint>
 #include <string>
 
+// Use likely, unlikely attributes from C++20 to improve performance of
+// reference counting when copy constructors are involved.
+#if defined(KOKKOS_ENABLE_CXX20)
+#if defined(KOKKOS_IMPL_REF_COUNT_BRANCH_LIKELY)
+#define KOKKOS_BRANCH_PROB [[likely]]
+#elif defined(KOKKOS_IMPL_REF_COUNT_BRANCH_UNLIKELY)
+#define KOKKOS_BRANCH_PROB [[unlikely]]
+#endif
+#endif
```

Comment:
Is there any reason why not just do this:

```suggestion
#if defined(KOKKOS_ENABLE_CXX20)
#define KOKKOS_BRANCH_PROB [[unlikely]]
#endif
```